pkg/git/repo.go

```diff
@@ -259,6 +260,15 @@ func (r repo) Clean() error {
 	return os.RemoveAll(r.dir)
 }
 
+// CleanPath deletes data in the given relative path in the repo with git clean.
+func (r repo) CleanPath(ctx context.Context, relativePath string) error {
+	out, err := r.runGitCommand(ctx, "clean", "-f", relativePath)
```

Comment:
Q: Since this is a relative path, do we need to check whether the given path points to an under dir or outer dir? (If it contains `../` or `/,` can the git clean command handle it carefully?)

Reply:
@khanhtc1202 
git clean command checks whether the path is under the repo or not.
So IMO, we don't need the check. WDYT?
```
~/oss/pipe-cd/pipecd [fujiwo]
% git clean -f ../tutorial                    
fatal: ../tutorial: '../tutorial' is outside repository at '$HOME/oss/pipe-cd/pipecd
```

Reply:
📝 I tried `git clean` at repo2 with both relative and absolute path.
```
at ~/test/repo2 
% git clean -f ../repo1/                                                         
fatal: ../repo1/: '../repo1/' is outside repository at '/Users/s14218/test/repo2'

at ~/test/repo2 
% git clean -f /Users/s14218/test/repo1                                          
fatal: /Users/s14218/test/repo1: '/Users/s14218/test/repo1' is outside repository at '/Users/s14218/test/repo2'
```

```
```
% tree .
.
├── repo1
│   └── README.md
├── repo2
       └── README.md
```
```

Reply:
Also added the test case for them [f1ef1a9](https://github.com/pipe-cd/pipecd/pull/5312/commits/f1ef1a996164b5ec5f054763fa629d7f33185d46)

Reply:
That's nice 👍